colossalai/shardformer/layer/normalization.py

```diff
@@ -34,17 +41,159 @@
     65536,
 ]
 
+class LayerNormBase(nn.Module):
 
-class FusedLayerNorm:
+    def named_parameters(self, prefix: str = '', recurse: bool = True) -> Iterator[Tuple[str, Parameter]]:
+        r"""Returns an iterator over module parameters, yielding both the
+        name of the parameter as well as the parameter itself.
+        """
+        gen = self._named_members(
+            lambda module: module._parameters.items(),
+            prefix=prefix, recurse=False)
+        for elem in gen:
```

Comment:
These methods are similar to those being defined in `torch.nn.Module`. Why write it again?

Reply:
The difference is that the overridden function removes the recursive processing of self._module. Otherwise, it may not correspond correctly with the keys of the Hugging Face model.

Reply:
You can simply set recursive to false for some functions. Extending the original class is a better idea.

Reply:
I changed the recurse parameter of named_parameters to its default value. However, for other functions, the recursive processing of submodules is not controlled by the recurse parameter. I created a base type here because the modifications made by FusedLayerNorm and FusedRMSNorm to these functions are the same.